jstreegrid.js

```diff
@@ -857,16 +934,30 @@
 			}
 			return result;
 		};
-		this._hide_grid = function (node) {
-			var children = node && node.children_d ? node.children_d : [], i;
-			// go through each column, remove all children with the correct ID name
-			findDataCell(this.uniq,children,this._gridSettings.gridcols).remove();
+    this._hide_grid = function (node) {
+      if (!node) { return true; }
+      this._detachColumns(node.id);
+      var children = node.children ? node.children : [], cols = this._gridSettings.columns || [], treecol = this._gridSettings.treecol;
+      // try to remove all children
+      for (var i = 0, len = children.length; i < len; i++) {
+        var child = this.get_node(children[i]);
+        // go through each column, remove all children with the correct ID name
+        for (var j = 0, lenj = cols.length; j < lenj; j++) {
+          if (j === treecol) { continue; }
+          findDataCell(this.uniq, child.id, j, $(this._domManipulation.columns[j])).remove();
+        }
+        if (child.state.opened) { this._hide_grid(child);}
+      }
+			
+      
+      this._reattachColumns(node.id);
 		};
 		this.holdingCells = {};
-		this.popHoldingCells = function (obj,col,hc) {
+    this.popHoldingCells = function (obj, col, hc) {
+      if (obj.state.hidden) { return $(); }
```

Comment:
Holding cells are just a mechanism to hold cells that need to be drawn, but cannot be drawn because their parents are not yet drawn.

Two points:

1. With the detach/attach, holding cells become less important. You basically can render all of them and then sort them. But the algorithm does get messy.
2. The grid used to be (several years ago) `div` and `span` elements appended to the end of the tree, so sorting, opening/closing, etc. was all handled as part of the tree. The downside was that getting horizontal positioning correct - let alone column rearrangement and resizing - was nearly impossible. So we switched to a true grid. Trade-offs.